File: js/055_JumpGame.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    var reach = 0, index = 0;
    while(index < nums.length && index <= reach){
        reach = Math.max(reach, index + nums[index]);
        if(reach >= nums.length-1) return true;
        ++index;
    }
    return index == nums.length;
};

File: cpp/055_JumpGame.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int reach = 0, index = 0;
        while(index < nums.size() && index <= reach){
            reach = std::max( reach, index + nums[index]);
            if(reach >= nums.size()-1) return true;
            ++index;
        }
        return index == nums.size();
    }
};